src/features/auth/lib/authMachine.ts

```diff
@@ -701,50 +779,59 @@ export const authMachine = createMachine<
           verificationUrl,
         };
       },
-      createFarm: async (context: Context, event: any): Promise<Context> => {
-        const { charityAddress, donation, captcha } = event as CreateFarmEvent;
+      createFarm: async (context: Context, event: any) => {
+        if (!context.user.rawToken) throw new Error("No token");
```

Comment:
What sort of use case would a player enter `createFarm` without a token?

Reply:
This is a safety measure in case the context was not set correctly.